src/Symfony/Component/Security/Core/User/OidcUserProvider.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\User;
+
+use Symfony\Component\Security\Core\Exception\UnsupportedUserException;
+
+use function Symfony\Component\String\u;
+
+/**
+ * OidcUserProvider creates an OidcUser from OidcUserBadge claims.
+ *
+ * @author Vincent Chalamon <vincentchalamon@gmail.com>
+ */
+class OidcUserProvider implements UserProviderInterface
```

Comment:
I don't think we should do this user provider approach. It won't work, as Symfony will never pass the extra `$claims` argument to it.

However, just before 6.2 stable, we added a way to completely by-pass user providers for exactly these situations. Let's use that instead: https://github.com/symfony/symfony/pull/48285
We need to do some groundwork first: allow user providers to be not configured (and null), see https://github.com/symfony/symfony/pull/48285#discussion_r1031450281 . I can take care of it this weekend.

Reply:
> It won't work, as Symfony will never pass the extra $claims argument to it.

It should, thanks to `OidcUserBadge::getUser`:
```php
$user = ($this->getUserLoader())($this->getUserIdentifier(), $this->claims);
```

As the OidcUserProvider is declared but not applied by default, the user has the following options:
1. declare its own UserProvider to load its own User object from the claims
2. use the default OidcUserProvider to load a default OidcUser object from the claims

Forcing the User in the UserBadge will break the 1st scenario.

Reply:
Ah, I missed that interesting trick. I'll dig in a bit more this weekend, as it didn't appear to work in my small test (and also try to explain my idea a bit better). Thanks for the detailed and quick response :)

Reply:
> I don't think we should do this user provider approach. It won't work, as Symfony will never pass the extra $claims argument to it.

I'm not sure we can avoid it actually, it's needed for user refreshment at least: 
https://github.com/symfony/symfony/blob/f3a9a0e2f22bd67074b596493a7e94779eb356e3/src/Symfony/Component/Security/Http/Firewall/ContextListener.php#L207 While we can make the Access Token authenticator pass that extra argument with an instanceof check

Reply:
We don't refresh users in stateless firewalls. Maybe I'm not understanding OIDC correctly, but afaik all access token based authenticators should be used with a stateless firewall, right?

Reply:
Indeed, OIDC in its regular form doesn't imply server-side sessions. But (AFAIK) it doesn't forbid it, and the access token authenticator currently does not enforce this. I'd personnaly agree to enforce it